samples/siwx91x_ota/src/main.c

```diff
@@ -0,0 +1,688 @@
+/*
+ * Copyright (c) 2025 Silicon Laboratories Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <string.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <ctype.h>
+#include <errno.h>
+#include <zephyr/net/wifi_mgmt.h>
+#include <zephyr/net/socket.h>
+#include <zephyr/net/http/client.h>
+#include "firmware_upgradation.h"
+#include "sl_wifi.h"
+
+/* Event masks */
+#define WIFI_SHELL_MGMT_EVENTS_COMMON                                                              \
+	(NET_EVENT_WIFI_SCAN_RESULT | NET_EVENT_WIFI_SCAN_DONE | NET_EVENT_WIFI_CONNECT_RESULT |   \
+	 NET_EVENT_WIFI_DISCONNECT_RESULT)
+
+/* Event flags */
+#define NET_IPV4_EVENT_READY BIT(1)
+#define NET_IPV6_EVENT_READY BIT(2)
+
+/* OTA update state machine states */
+enum ota_state {
+	OTA_STATE_SCAN,           /**< Scan for WiFi networks */
+	OTA_STATE_SERVER_CONNECT, /**< Connect to OTA server */
+	OTA_STATE_DOWNLOAD        /**< Download and process firmware image */
+};
+
+struct http_parse_data {
+	bool is_tls_enabled;
```

Comment:
`is...enabled` is long. You can just call it `use_tls` and `have_tls` with keeping meaningfulness.